### PR TITLE
Standardize `Range*` documentation

### PR DESCRIPTION
**Pull request by czipperz**
This updates the final example in the documentation for the types
`Range`, `RangeFrom`, `RangeFull`, `RangeInclusive`, `RangeTo`,
`RangeToInclusive`.

### CODE REVIEW

File: src/libcore/ops/range.rs [hunk 1]
```diff
@@ -60,11 +62,13 @@ impl fmt::Debug for RangeFull {
...
-/// assert_eq!(arr[ ..3], ['a', 'b', 'c',    ]);
-/// assert_eq!(arr[1.. ], [     'b', 'c', 'd']);
-/// assert_eq!(arr[1..3], [     'b', 'c'     ]);  // Range
+/// let arr = [0, 1, 2, 3, 4];
+/// assert_eq!(arr[ ..  ], [0,1,2,3,4]);
+/// assert_eq!(arr[ .. 3], [0,1,2    ]);
+/// assert_eq!(arr[ ..=3], [0,1,2,3  ]);
+/// assert_eq!(arr[1..  ], [  1,2,3,4]);
+/// assert_eq!(arr[1.. 3], [  1,2    ]);
+/// assert_eq!(arr[1..=3], [  1,2,3  ]);  // Range
```
Comment (shepmaster) [inline src/libcore/ops/range.rs]: This is not `Range`.

**Review by shepmaster — CHANGES_REQUESTED** [review]

**Comment by czipperz** [comment]
@shepmaster Fixed that.  Thanks for the catch

**Comment by shepmaster** [comment]
Thanks!

**Review by shepmaster — APPROVED** [review]

**Pull request merged**